### PR TITLE
Add escapes to CMake generator for cpp_info variables

### LINKED ISSUE #2554: CMake generator creates invalid CMake files if cflags/linker flags have backslashes
To help us debug your issue please explain:

- [X] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [X] I've specified the Conan version, operating system version and any tool that can be relevant.
- [X] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Conan 1.2.0-dev

Generated CMake files are not properly escaped if c(pp)flags/linkflags contain backslashes. This results in a CMake error when a flag requires this inside of it. The package cannot perform the escape itself because it cannot assume the consumer will use CMake (a non-CMake consumer would need to un-escape the flags, in that case).

This is similar to the problem already fixed in #1640 but in a different place.

I already have a fix for this issue; I will create an appropriate PR. However, this change is possibly breaking because packages running into this issue may have assumed their consumers use CMake and worked around it by manually escaping inside of cpp_info. However, I have made a cursory check on conan-center and my sample did not encounter any packages running into this problem.

### PR DESCRIPTION
**Pull request by mpdelbuono**
- [X] Refer to the issue that supports this Pull Request.
- [X] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [X] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [X] I've followed the PEP8 style guides for Python code.
- [X] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

Fixes #2554 

CMake generator now properly escapes variables in the generated cmake file. Includes associated test. The test mimics the use case (using a path name inside of a cpp flag).

### CODE REVIEW

File: conans/client/generators/cmake.py [hunk 1]
```diff
@@ -7,33 +7,42 @@
 
 class DepsCppCmake(object):
     def __init__(self, cpp_info):
+        def escape(value, path=False):
+            """Escapes the specified string for use in a CMake command surrounded with double quotes
+                :param value the string to escape
+                :param path True if this should be treated as a path (backslash will be transformed to forward slash)
+                                 or False otherwise (backslash will be transformed to double backslash)"""
+            return '{0}'.format(value.replace('\\', '/' if path else '\\\\')
```
Comment (lasote) [inline conans/client/generators/cmake.py]: Could you provide an example of a non path value with backslash? Any flag?

**Comment by lasote** [comment]
Many thanks for your contribution!
I would like to see an integration test (using real cmake) with some flag containing the quotes and the $ to verify that CMake is working ok (and to be able to confirm manually that CMake was failing otherwise)

**Comment by memsharded** [comment]
Verified that this fails. I am merging and adding myself an integration test.

Thanks @mpdelbuono for fixing this!

**Comment by Nipheris** [comment]
@memsharded Do we need a separate issue for `CMakeToolchain`? It also lacks escaping for variables (for example, [here](https://github.com/conan-io/conan/blob/1.42.2/conan/tools/cmake/toolchain.py#L77)), and suffers from the same problem.

**Comment by memsharded** [comment]
Hi @Nipheris 
yes, definitely, if CMakeToolchain is failing because of this, please submit a new issue with details to reproduce, and we will try to fix it

**Pull request merged**